Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Search.hs

```diff
@@ -131,52 +133,57 @@ handler merchant sReq = do
           QSearchRequestSpecialZone.create searchRequestSpecialZone
-        now <- getCurrentTime
-        listOfSpecialZoneQuotes <- do
-          allFarePolicies <- FarePolicyS.findAllByMerchantId merchant.id
-          let farePolicies = selectFarePolicy result.distance allFarePolicies
-          let listOfVehicleVariants = listVehicleVariantHelper farePolicies
-          for listOfVehicleVariants $ \farePolicy -> do
-            fareParams <-
-              calculateFareParameters
-                CalculateFareParametersParams
-                  { farePolicy = farePolicy,
-                    distance = result.distance,
-                    rideTime = sReq.pickupTime,
-                    waitingTime = Nothing,
-                    driverSelectedFare = Nothing,
-                    customerExtraFee = Nothing
-                  }
-            buildSpecialZoneQuote
-              searchRequestSpecialZone
-              fareParams
-              merchant.id
-              result.distance
-              farePolicy.vehicleVariant
-              result.duration
-        Esq.runTransaction $
-          for_ listOfSpecialZoneQuotes QQuoteSpecialZone.create
-        return (Just (mkQuoteInfo fromLocation toLocation now <$> listOfSpecialZoneQuotes), Nothing)
+        flowType <- getFlowType merchantId fromLocationLatLong toLocationLatLong
+        case flowType of
+          DFareProduct.RIDE_OTP -> do
+            now <- getCurrentTime
+            let listOfVehicleVariants = listVehicleVariantHelper farePolicies
+            listOfSpecialZoneQuotes <- do
+              for listOfVehicleVariants $ \farePolicy -> do
+                fareParams <-
+                  calculateFareParameters
+                    CalculateFareParametersParams
+                      { farePolicy = farePolicy,
+                        distance = result.distance,
+                        rideTime = sReq.pickupTime,
+                        waitingTime = Nothing,
+                        driverSelectedFare = Nothing,
+                        customerExtraFee = Nothing
+                      }
+                buildSpecialZoneQuote
+                  searchRequestSpecialZone
+                  fareParams
+                  merchant.id
+                  result.distance
+                  farePolicy.vehicleVariant
+                  result.duration
+                  mbSpecialLocationTag
+            Esq.runTransaction $
+              for_ listOfSpecialZoneQuotes QQuoteSpecialZone.create
+            return (Just (mkQuoteInfo fromLocation toLocation now <$> listOfSpecialZoneQuotes), Nothing)
+          DFareProduct.NORMAL -> do
+            driverPoolCfg <- getDriverPoolConfig merchantId result.distance
+            estimateInfos <- buildEstimatesInfos fromLocation toLocation driverPoolCfg result farePolicies mbSpecialLocationTag
+            return (Nothing, Just estimateInfos)
       else do
         driverPoolCfg <- getDriverPoolConfig merchantId result.distance
```

Comment:
can you avoid this repetition of code in NORMAL flow type and else part, both seems to be same only.

Reply:
Yes, Done.